COMPOUND_COMPONENTS.md

```diff
@@ -0,0 +1,337 @@
+# Compound Components
+
+## What is a compound component?
+
+Compound components is a pattern in which components are used together in such a way that describes a higher level of abstraction. A desktop computer is composed of many components that can be serviced and configured individually. Similarly, a compound component allows simultaneous configuration through access to each sub-component. Here are the parts of a compound component:
+
+* Container component
+* Sub-components
+* Shared model (optional)
+* Behavior hooks (optional)
+
+A compound component contrasts with a configuration component which instead configures from a single interface. A configuration component might be like choosing a desktop computer based on stats - how much RAM or how fast the CPU should be or based on what you want to use it for. A compound component is more like buying the parts individually and assembling yourself.
+
+Configuration component:
+```tsx
+<Tabs items={[
+  { title: 'First Tab', content: 'First Tab Contents' },
+  { title: 'Second Tab', content: 'Second Tab Contents' },
+]} />
+```
+
+Compound component:
+```tsx
+<Tabs>
+  <Tabs.List>
+    <Tabs.Item>First Tab</Tabs.Item>
+    <Tabs.Item>Second Tab</Tabs.Item>
+  </Tabs.List>
+  <Tabs.Panel>First Tab Contents</Tabs.Panel>
+  <Tabs.Panel>Second Tab Contents</Tabs.Panel>
+</Tabs>
+```
+
+In this example, `Tabs` is the container component and `Tabs.List` is a sub-component.
+
+Some compound components might not contain state or behavior. An example might be an `IconButton` which is a button that contains an icon. It might be a compound component only for styling purposes, but doesn't contain any special state or behaviors:
+
+```tsx
+<IconButton onClick={onClick}>
+  <IconButton.Icon icon={icon} />
+  <IconButton.Text>Button Text</IconButton.Text>
+</IconButton>
+```
+
+## Container Components
+
+A container component is the entry point to a compound component. A container component could represent a real DOM element, or be a non-element container. For example, the `Pagination` component has a container component that represents a `role=nav` element. The `Tabs` container component, however, does not contain a semantic element.
+
+If a compound component contains any state or behavior, it will also provide a shared model to sub-components via [React context](https://reactjs.org/docs/context.html). A container component takes props for either the model or configuration for the model. In the `Tabs` compound component example, it might look like this:
+
+```tsx
+const TabsModelContext = React.createContext({})
```

Comment:
In building a new component, `useTabsModel` and `useTabsModelContext` are very close to each other and return the same type. If you use `useTabsModel` (first autocomplete suggestion) instead of the context version, you'll get a new model instance rather than the shared reference and things don't work the way you expect.

The suggestion is instead to export the `TabsModel` as a context reference and don't export a `useTabsModelContext` hook. This also helps reinforce this is explicitly React Context and not a new model.